plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/catalog/snowflake/TestIcebergSnowflakeCatalogConnectorSmokeTest.java

```diff
@@ -688,8 +689,12 @@ public void testSnowflakeNativeTable()
     @Override
     public void testIcebergTablesFunction()
     {
-        assertThatThrownBy(super::testIcebergTablesFunction)
-                .hasMessageContaining("schemaPath is not supported for Iceberg snowflake catalog");
+        String schemaTablesValues = "VALUES " + getQueryRunner()
+                .execute("SELECT table_schema, table_name FROM iceberg.information_schema.tables WHERE table_schema='%s'".formatted(SNOWFLAKE_TEST_SCHEMA.toLowerCase(ENGLISH)))
+                .getMaterializedRows().stream()
+                .map(row -> "('%s', '%s')".formatted(row.getField(0), row.getField(1)))
+                .collect(joining(", "));
+        assertQuery("SELECT * FROM TABLE(iceberg.system.iceberg_tables(SCHEMA_NAME => '%s'))".formatted(SNOWFLAKE_TEST_SCHEMA.toLowerCase(ENGLISH)), schemaTablesValues);
```

Comment:
I think this test is flaky. What happens if a new object is created after listing `iceberg.information_schema.tables` and before executing `iceberg_tables` function?

Reply:
That was the reason the generic test creates its own schemas but in this case, it is not needed because this test does not create iceberg tables other than potentially during setup.
There is one test that creates a native table but that will not be listed by `TrinoSnowflakeCatalog#listTables`